### PR TITLE
xSDK Changes

### PR DESCRIPTION
**Pull request by Angelyr**
- Doxygen page
- Smoke test
- optional print statement
- print version

### CODE REVIEW

File: Doxyfile [hunk 1]
```diff
@@ -0,0 +1,2736 @@
...
+
+DOXYFILE_ENCODING      = UTF-8
+
+# The PROJECT_NAME tag is a single word (or a sequence of words surrounded by
+# double-quotes, unless you are using Doxywizard) that should identify the
+# project for which the documentation is generated. This name is used in the
+# title of most generated pages and in a few other places.
+# The default value is: My Project.
+
+PROJECT_NAME           = "Pcms"
```
Comment (cwsmith) [inline Doxyfile:45]: Should this be "PCMS"?

File: src/pcms/print.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  FILE* getStderr() { return pp_stderr; }
+
+  void setStdout(FILE* out) {
+    assert(out != NULL);
+    pp_stdout = out;
+  }
+
+  void setStderr(FILE* err) {
+    assert(err != NULL);
+    pp_stderr = err;
```
Comment (cwsmith) [inline src/pcms/print.cpp:18]: Is the `pp_` prefix from PUMIPic?

File: test/CMakeLists.txt
Comment (cwsmith) [inline test/CMakeLists.txt:1]: The testing 'helper' macros were moved to `testing.cmake`.  Was the intention to also move all the tests (i.e. the calls to functions like `dual_mpi_test(...)` and their supporting code) to `testing.cmake`?  I don't see `testing.cmake` included/referenced in the CMakeLists.txt file but may have missed it.

**Review by cwsmith — CHANGES_REQUESTED** [review]
Thank you.  A few comments are below.

File: test/CMakeLists.txt
Comment (Angelyr) [reply test/CMakeLists.txt:1]: I moved those functions so I could reuse them for the smoke tests. I included the file in the main CMakeLists.txt file and that is also where I created the smoke test. That way the smoke test is still created when testing is disabled.

Comment (cwsmith) [reply test/CMakeLists.txt:1]: OK.  Sounds good.  Thank you.

**Review by cwsmith — APPROVED** [review]

**Review by jacobmerson — APPROVED** [review]

**Comment by jacobmerson** [comment]
Thanks!

**Pull request merged**